pkg/clusterclientstore/store.go

```diff
@@ -332,5 +341,11 @@ func buildConfig(host string, secret *corev1.Secret, restTimeout *time.Duration)
 		config.KeyData = key
 	}
 
-	return config
+	if insecureSkipTlsVerify, err := util.GetBool(secret, "tls.insecure-skip-tls-verify"); err != nil && util.IsDecodeError(err) {
```

Comment:
`insecureSkipTlsVerify` is a `bool`, and `util.GetBool` returns `false` in case an error is returned. The check on `IsDecodeError` is meant to catch cases where the value exists in the secret but couldn't be decoded (`base64.StdEncoding.DecodeString` has failed) or parsed (`strconv.ParseBool` has failed).